core/vm/contracts.go

```diff
@@ -1182,3 +1193,27 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 
 	return h
 }
+
+// txIndex implements EIP-7793 TXINDEX precompile.
+type txIndex struct {
+	indexFn func() int
```

Comment:
IMO this is not the best name, we already know from the type that this is function, I'd rename this to `txIndex` or `getTxIndex` or smth. like that :)